colour/colorimetry/spectrum.py

```diff
@@ -1939,8 +2050,49 @@ def align(self,
                 77.18     ...,  77.18     ...,  77.18     ...,  77.18     ...])
         """
 
-        self.interpolate(shape)
-        self.extrapolate(shape, method, left, right)
+        self.interpolate(shape, interpolation_method)
+        self.extrapolate(shape,
+                         extrapolation_method,
+                         extrapolation_left,
+                         extrapolation_right)
+
+        return self
+
+    def trim(self, shape):
```

Comment:
Not sure. I think the name should make clear that this is about `shape`, not values.
Not as concise, but what about `trim/cull/clip_shape`?